python/ray/train/_internal/storage.py

```diff
@@ -463,6 +463,13 @@ def __init__(
             self.storage_path, storage_filesystem
         )
 
+        if not self.storage_filesystem or isinstance(
+            self.storage_filesystem, pyarrow.fs.LocalFileSystem
+        ):
+            self.storage_local_path = self.storage_path
+        else:
+            self.storage_local_path = default_results_dir
+
```

Comment:
@justinvyu lmk what you think. Previously the `storage_local_path` was always the ray_results directory. But I think we should actually set it to the local path if a local path was set. This also matches previous behavior.

Reply:
I'm ok with this. @ericl did not want ANY parsing of the storage path (i.e detecting if it's remote or not).

But, because this is AFTER `pyrarrow.fs.from_uri`, and we're just doing a typecheck of the pyarrow fs, so I think this is ok.

The difference is that previously, we did some checks on the raw uri that was being passed by the user.

This will also remove the need for us to either re-introduce some kind of `local_dir` param or make the environment variable more visible.
`storage_path` is the 1 way to set the experiment path.